File: src/trees/same_tree.py
# Definition for a binary tree node.
from typing import Optional

from src.trees.utils import TreeNode


class Solution:
    def isSameTree(self, rootNode1: Optional[TreeNode], rootNode2: Optional[TreeNode]) -> bool:
        if rootNode1 is None and rootNode2 is None:
            return True
        if rootNode1 is None or rootNode2 is None:
            return False
        return rootNode1.val == rootNode2.val \
               and self.isSameTree(rootNode1.left, rootNode2.left) \
               and self.isSameTree(rootNode1.right, rootNode2.right)

File: java/src/main/java/com/blind75/trees/SameTree.java
package com.blind75.trees;

public class SameTree {
    public boolean isSameTree(TreeNode treeNode1, TreeNode treeNode2) {
        if (treeNode1 == null && treeNode2 != null) {
            return false;
        }
        if (treeNode1 != null && treeNode2 == null) {
            return false;
        }
        if (treeNode1 == null && treeNode2 == null) {
            return true;
        }
        return treeNode1.val == treeNode2.val
                && isSameTree(treeNode1.left, treeNode2.left)
                && isSameTree(treeNode1.right, treeNode2.right);
    }
}
